src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -17,124 +21,145 @@ namespace PedestrianPlugin
 
-    // Run the I/O service. The call will return when
-    // the socket is closed.
+    // Run the I/O service. The call will return when the socket is closed.
     ioc.run();
 
-	return EXIT_SUCCESS;
+    return EXIT_SUCCESS;
 }
 
-int PedestrianPlugin::checkXML()
+void PedestrianPlugin::StopWebSocket()
 {
-	//first xml will be empty string
-	std::string lastGeneratedXML = "";
+    if (flirSession && runningWebSocket)
+    {
+        PLOG(logDEBUG) << "Stopping WebSocket session";
+		beast::error_code ec;
+        flirSession->on_close(ec);
+        runningWebSocket = false;
+    }
+    else
+    {
+        PLOG(logDEBUG) << "WebSocket session was not running or already stopped.";
+    }
+}
 
+void PedestrianPlugin::checkXML()
+{
 	//if a new psm xml has been generated the FLIR web socket, send it to the BroadcastPSM function
 	while (true)
 	{
 		if (flirSession == nullptr)
 		{
-			PLOG(logDEBUG) << "flir session not yet initialized: " << std::endl;
+			PLOG(logDEBUG) << "flir session not yet initialized: ";
 		}
 		else
 		{	
 			//retrieve the PSM queue and send each one to be broadcast, then pop		
-			std::queue<string> currentPSMQueue = flirSession->getPSMQueue();
+			std::queue<std::string> currentPSMQueue = flirSession->getPSMQueue();
 
 			while(!currentPSMQueue.empty())
 			{		
-				char* char_arr = &currentPSMQueue.front()[0];
+				const char* char_arr = &currentPSMQueue.front()[0];
 
 				BroadcastPsm(char_arr);
 				currentPSMQueue.pop();
 			}			 
 		}
-		
 	}	
-	return EXIT_SUCCESS;
 }
 
-int PedestrianPlugin::StartWebService()
+void PedestrianPlugin::PedestrianRequestHandler(QHttpEngine::Socket *socket)
 {
-	//Web services 
-	char *placeholderX[1]={0};
-	int placeholderC=1;
-	QCoreApplication a(placeholderC,placeholderX);
-
- 	QHostAddress address = QHostAddress(QString::fromStdString (webip));
-    quint16 port = static_cast<quint16>(webport);
+	QByteArray st;
+	while(socket->bytesAvailable() > 0)
+	{	
+		auto readBytes = socket->readAll();
+		st.append(readBytes);
+	}
 
+	if(st.size() == 0)
+	{
+		PLOG(logERROR) << "Received PSM is empty and skipped.";
+		socket->setStatusCode(QHttpEngine::Socket::BadRequest);
+		socket->writeHeaders();
+		socket->close();
+		return;
+	}
+	PLOG(logINFO) << "Received PSM bytes size: " << st.size();
 
-	QSharedPointer<OpenAPI::OAIApiRequestHandler> handler(new OpenAPI::OAIApiRequestHandler());
-	handler = QSharedPointer<OpenAPI::OAIApiRequestHandler> (new OpenAPI::OAIApiRequestHandler());
+	std::string psmMsgdef = st.data();
+	std::list<std::string> psmSL = {};
+	psmSL.push_back(psmMsgdef);
 
-    QObject::connect(handler.data(), &OpenAPI::OAIApiRequestHandler::requestReceived, [&](QHttpEngine::Socket *socket) {
+	// Catch parse exceptions
+    try {
+		for(const auto& psm_s: psmSL)
+		{
+			BroadcastPsm(psm_s);
+			socket->setStatusCode(QHttpEngine::Socket::Created);
+		}
+	}
+	catch(const J2735Exception &e) {
+        PLOG(logERROR) << "Error parsing file: " << e.what();
```

Comment:
**NIT** : Is the indentation off in this file or is github just doing funny stuff.